src/algorithms/tonal/pitchyinfft.cpp

```diff
@@ -189,8 +245,9 @@ void PitchYinFFT::compute() {
       yinMin = -_amplitudes[0];
     }
     else {
-      // TODO this should never happen, but some people reported it happening in their real time applications.
-      throw EssentiaException("PitchYinFFT: it appears that no peaks were found by PeakDetection. If you read this message, PLEASE, report this issue to the developers with an example of audio on which it happened.");
+      tau = 0.0; // it will provide zero-pitch and zero-pitch confidence.
```

Comment:
This change is out of scope of weighting functions.  At least would should add the description of this change to the PR.

Reply:
Oh yes! this is something we need in the ODISEI project to avoid a crucial error. Indeed we would need to launch a negative pitch value. No problem I will redo what essentia does in this cases.

Reply:
This change had been already included in the merge.